Update predicted/expected to reference/modeled

lboeman opened the pull request:
closes #194

alorenzo175 commented on dashboard/src/utils/schemaIndexing.ts:
```diff
@@ -23,7 +23,7 @@ export function indexSystemFromSchemaPath(
   // Pull first index, so the "" that exists for the root component does not
   // fail, and when we have a valid non-falsy index we'll traverse the system
-  // as expected.
+  // as modeled.
```
```suggestion
  // as expected.
```

alorenzo175 commented on dashboard/tests/unit/jobs/test_csvmapper.spec.ts:
```diff
@@ -152,15 +152,15 @@ describe("Test CSV Mapper", () => {
 
     const inverter = testJob.definition.system_definition.inverters[0];
     const array = inverter.arrays[0];
-    const expectedMetadata = {
```
doesn't really matter, but was this one (and ln 163) supposed to change?

lboeman replied:
No, thanks.

alorenzo175 commented on dashboard/tests/unit/test_auth.spec.ts:
```diff
@@ -169,7 +169,7 @@ describe("test user", () => {
       updated_at: "2222-22-22T22:22:22-10:00"
     };
-    const expected = {
+    const modeled = {
```
```suggestion
    const expected = {
```

alorenzo175 commented on dashboard/tests/unit/test_auth.spec.ts:
```diff
@@ -179,7 +179,7 @@ describe("test user", () => {
     };
     const instance = new User(user);
-    expect(instance).toEqual(expected);
+    expect(instance).toEqual(modeled);
```
```suggestion
    expect(instance).toEqual(expected);
```

alorenzo175 commented on dashboard/tests/unit/utils/test_unit_conversion.spec.ts:
```diff
@@ -2,66 +2,66 @@ import { getUnitConverter } from "@/utils/unitConversion";
 describe("Test power conversion", () => {
   it("kW => W", () => {
-    const expected = [
+    const modeled = [
```
change all these back to expected in this file

alorenzo175 reviewed (commented):
Mostly looks good to me. I can't test everthing on my laptop though, so should definitely double check on dev after merged. One more change is probably "original" to "reference" i.e. in "Upload original weather data"

Still a few missing changes in :
JobHandler.vue lns 194, 168, 181
JobParams.vue ln 47
CompareJobsParams.vue want to change "containsPredicted"?

Need to update openapi.json and run the migration and add db/schema.sql to git

lboeman commented:
Alright, cleaned up the remaining unwanted expected->modeled replacements and replaced "original" in data types with "reference" as well as updated the `schema.sql`. For future reference, I added a note to the `db/README` about installing `mysqldump`. I found that since I'm running docker in a container and dont have the mysqldump installed locally, the `dbmate up`  failed to update the schema and did so silently.

wholmgren commented:
@cwhanse can you confirm this is correct? I think it makes sense, but the terminology has tripped me too many times to be confident.

> One more change is probably "original" to "reference" i.e. in "Upload original weather data"

wholmgren commented on dashboard/src/components/jobs/parameters/JobParams.vue:
```diff
@@ -10,8 +10,8 @@
             <!-- For selecting between diferent usecase variants e.g. CompareJobParams
-                 Allows user to select betweel predicted-actual, expected-actual, and
-                 predicted expected.    
+                 Allows user to select betweel reference-actual, modeled-actual, and
```
between

wholmgren commented on dashboard/src/components/jobs/parameters/JobParams.vue:
```diff
@@ -10,8 +10,8 @@
-                 Allows user to select betweel predicted-actual, expected-actual, and
-                 predicted expected.    
+                 Allows user to select betweel reference-actual, modeled-actual, and
+                 reference modeled.
```
reference-modeled

cwhanse commented:
> @cwhanse can you confirm this is correct? I think it makes sense, but the terminology has tripped me too many times to be confident.
> 
> > One more change is probably "original" to "reference" i.e. in "Upload original weather data"

+1 to changing "original" to "reference"

wholmgren reviewed (commented):
and reviewed the diff and searched for occurrances of these words. looks good.
